packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -2415,6 +2415,17 @@ class TextSelectionGestureDetectorBuilder {
     if (delegate.selectionEnabled) {
       switch (defaultTargetPlatform) {
         case TargetPlatform.iOS:
+          // Show the floating cursor
+          SchedulerBinding.instance.addPostFrameCallback((Duration duration) {
+            final RawFloatingCursorPoint cursorPoint = RawFloatingCursorPoint(
+              state: FloatingCursorDragState.Start,
+              localPosition: renderEditable.globalToLocal(details.globalPosition),
+              offset: Offset.zero,
+            );
+            editableText.updateFloatingCursor(cursorPoint);
+          }, debugLabel: 'TextSelectionGestureDetectorBuilderDelegate.updateFloatingCursor');
+          continue macOSCase;
```

Comment:
It will probably be easier if we update the logic in `RenderEditable.updateFloatingCursor` to update the selection immediately on `FloatingCursorDragState.Start` (as all types of floating cursors seem to relocate the caret to where the floating cursor is on start?), so we don't have to do this?

Reply:
Good point. Then does it make sense to add the new text selection to the `RawFloatingCursorPoint` class? 
```dart
class RawFloatingCursorPoint {
  // (local offset, new caret location)
  //  Only non-null when starting a floating cursor via long press.
  final (Offset, TextPositon)? startLocation;
}
```

Then in editable_text.dart we can use that information in the `FloatingCursorDragState.Start` clause if non-null. It's error-prone to use `addPostframeCallback` in general since you don't know what is going to happen in that frame (the render object can be disposed, for example).